### ISSUE TITLE
Package structure

### ISSUE THREAD

**Issue by sometimes-i-send-pull-requests** [sbyt32/Preordain#16]
Right now, your code has no overarching package. You have modules at the top level and a couple of directories. The directories are also named like you are organizing files, not so much like you are organizing modules and packages.

Consider introducing an overarching package that you move your code into. Then consider the current packages you have and figure out where they would fit in this hierarchy.

I am not sure how to describe how to pick better names. I have ideas, but I don't want to be prescriptive. Let me know if you have specific questions.

**Comment by sbyt32** [comment]
I'm not too sure what you mean here, if you can show a small example either via a diagram or something, that'd help a lot.

**Comment by sometimes-i-send-pull-requests** [comment]
Here is an example: https://stackoverflow.com/a/64987404

See how everything is inside a top-level project package. In addition, see how the subpackages are named like packages, not like you are organizing files (e.g. “api_files” would only be appropriate if you were doing an API about files). See also how they are not nested too deeply. Currently, your fully qualified names are unusually long.

**Comment by sbyt32** [comment]
After doing some digging around in [some](https://fastapi.tiangolo.com/tutorial/bigger-applications/#an-example-file-structure) [other](https://github.com/zhanymkanov/fastapi-best-practices#15-backgroundtasks--asynciocreate_task) places too, I think I have an idea as to how to restructure my project, with SQLAlchemy in mind. Keep in mind it might just be incredibly wrong.
```

│ price_fetch
├─ src
│   ├─ card_price_db
│   │   ├─ __init__.py
│   │   ├─ router.py
│   │   ├─ schema.py # pydantic models
│   │   ├─ models.py # database models
│   │   ├─ dependencies.py
│   │   ├─ config.py
│   │   ├─ utils.py # various functions
│   │   ├─ exceptions.py
│   ├─ docs
│   │   ├─ *.md  # function explanations, notes, etc
│   ├─ __init__.py
│   ├─ config.py
│   ├─ models.py
│   ├─ exceptions.py
│   ├─ database.py
│   └─ exceptions.py
├─ tests/
│   └─ card_price_db
│
├─ .gitignore
├─ README.md
├─ requirements.txt
├─ logger.py # I feel like this can be moved somewhere else
├─ set_up.py # I feel like this can be moved somewhere else
```

My only question is: _Where do I put those data scraping scripts?_ I know I mentioned that I'd like to start working towards pulling bulk data and not just grabbing a few cards, but that still would leave me with wondering where I put those scripts. Should I abandon them for now and make that into a completely different setup?

**Comment by sometimes-i-send-pull-requests** [comment]
1. Although it seems like the reference you found used it, I would not recommend using `src` as a package name. It is confusing and I can’t think of any other code that uses that convention — as a directory, yes, but not as a package name (subtle distinction; but if you have `from src…` or `import src…` then something is wrong). I suggest picking a different top-level package name related to your project.

    The reference you found has this to say about why they chose it:

    > Structure that I found more scalable and evolvable is inspired by Netflix's [Dispatch](https://github.com/Netflix/dispatch) with some little modifications.

    Their “little modifications” appear to be mistakes/misunderstandings to me, and I suggest not copying their mistakes. Take a look at the Dispatch code they reference: All the code is under src/dispatch, and all the imports say “from dispatch…” or “import dispatch…”, with no mention of src. This is correct.

    “dispatch” is an great top-level package name. Try to come up with one.

2. You can put the scraping code in a `scraping` subpackage. For example, if this was a part of Dispatch, maybe you would have a `dispatch.scraping.price_data_fetcher`.

**Comment by sbyt32** [comment]
Testing the following structure in 0cdaa6f

```
│ mtg_price_fetch
├─ preordain
│   ├─ groups
│   │   ├─ __init__.py
│   │   ├─ models.py
│   │   ├─ router.py
│   │   ├─ utils.py (Optional)
│   ├─ information
│   │   ├─ __init__.py
│   │   ├─ models.py
│   │   ├─ router.py
│   │   ├─ utils.py (Optional)
│   ├─ inventory
│   │   ├─ __init__.py
│   │   ├─ models.py
│   │   ├─ router.py
│   │   ├─ utils.py (Optional)
│   ├─ price_sales
│   │   ├─ __init__.py
│   │   ├─ models.py
│   │   ├─ router.py
│   │   ├─ utils.py (Optional)
│   ├─ utils
│   ├─ api.py
│   ├─ dependencies.py
│   ├─ models.py
│   ├─ exceptions.py
├─ tests
├─ docs
├─ logs
├─ set_up.py (Restructure to be a class rather than a bunch of functions?)
├─ data_scraping (Not sure if here or not)
```

Some things are still in the process of being moved around, but what do you think?

**Comment by sometimes-i-send-pull-requests** [comment]
That looks great to me!

**Comment by sbyt32** [comment]
Merged #24
